serverless/serverless
# AWS: Throw an error if ${ssm} fail to populate value, as same as ${cf} and ${s3}

## What did you implement:

Closes #4447
Closes #5757
Closes #5781

## How did you implement it:

Before this PR, `${ssm}` catches `400` error from AWS SSM, and returns `undefined` as fulfilled value.
This may cause *broken* service deployment and user may not notice that.
And, such behavior is incosistent with `${cf}` and `${s3}`.

After this PR, `${ssm}` aborts whenver SSM returns error.
The error can be recovered with the fallback value in the form of `${ssm:/path/param, "fallback"}` (already implemented).
So every AWS-specific variable syntaxes has uniform behavior.

## How can we verify it:

1.  Install the patched serverless from branch.

```bash
npm install -g exoego/serverless#ssm-should-fail`
sls create -t aws-nodejs --path ssm-fail
cd ssm-fail
```

2. Update serverless.yml with 

```yml
service:
  name: aws-nodejs

provider:
  name: aws
  runtime: nodejs8.10

functions:
  hello:
    handler: handler.hello

custom:
  ssmVar: ${ssm:/path/to/service/myParamS}
```

3. Run `sls deploy` and confirm it aborts like
 
```bash
$ sls deploy

  Serverless Error ---------------------------------------
 
  Error getting value for ssm:/path/to/service/myParamS. ParameterNotFound
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           8.10.0
     Serverless Version:     1.36.3
```


4. Add fallback value to serverless.yml like `ssmVar: ${ssm:/path/to/service/myParamS, 'fallback'}`
5. Run `sls deploy` and confirm it complete successfully 

## Todos:

- [x] <del>Write documentation</del> N/A

***Is this ready for review?:*** YES
***Is it a breaking change?:*** COULD BE
